### ISSUE TITLE
Async HTTP request fails if first one gets a cancled token.

### ISSUE THREAD

**Issue by hajimeku** [Benedicht/BestHTTP-Issues#10]
Hey!

So what happens is that the "HTTPUpdateDelegator" does not exist if the first request is given a canceled cancelation token.

`
CancellationTokenSource cancellationTokenSource = new CancellationTokenSource();
CancellationToken cancellationToken = cancellationTokenSource.Token;
cancellationTokenSource.Cancel();
var response = await httpRequest.GetHTTPResponseAsync(cancellationToken);
`
The task never completes because it doesn't get to:
public static HTTPRequest SendRequest(HTTPRequest request) and calls 
            if (!IsSetupCalled)
                Setup();

I fixed it now by calling Setup in:
        ```
[UnityEngine.RuntimeInitializeOnLoadMethod(UnityEngine.RuntimeInitializeLoadType.AfterSceneLoad)]
        static void Reset()
        {
            IsSetupCalled = false;
            HTTPManager.Logger.Information("HTTPManager", "Reset called!");
HTTPManager.Setup();
        }
```

im using the 2.0.5 version, I did see in release notes 2.0.6 but I cant download it yet.

**Comment by Benedicht** [comment]
@hajimeku 

Thanks for the report!
I think the main problem was that calling with a canceled token no callback is called because the request gets aborted before it tried to call its Send.

v2.0.6 isn't released yet, i just uploaded it for review in the asset store. When they accept it in 1-3 days it's going to be released and available automatically.
If you send me a mail though, i can send a new fixed package.

**Comment by hajimeku** [comment]
Is this fixed in v2.0.6?

The main problem is that the callback is not called because the HTTPUpdateDelegator is not spawned yet and does not process the state change. if any request has callend "Send"it works fine.

**Comment by Benedicht** [comment]
@hajimeku 

No, v2.0.6 isn't fixes this, but if you send me a mail (besthttp@gmail.com) than i can send a new package that should fix it.
